File: backend/Python/app/features/pizzas/repository.py
from dataclasses import dataclass
from typing import Dict, Optional
from uuid import uuid4


@dataclass
class Pizza:
    id: str
    name: str
    price: float


class PizzaRepository:
    async def create(self, name: str, price: float) -> Pizza:
        raise NotImplementedError

    async def get_by_id(self, pizza_id: str) -> Optional[Pizza]:
        raise NotImplementedError


class InMemoryPizzaRepository(PizzaRepository):
    def __init__(self) -> None:
        self._store: Dict[str, Pizza] = {}

    async def create(self, name: str, price: float) -> Pizza:
        pizza = Pizza(id=str(uuid4()), name=name, price=price)
        self._store[pizza.id] = pizza
        return pizza

    async def get_by_id(self, pizza_id: str) -> Optional[Pizza]:
        return self._store.get(pizza_id)


pizza_repository: PizzaRepository = InMemoryPizzaRepository()

File: backend/Node/src/features/pizzas/repository.ts
export type Pizza = {
	id: string;
	name: string;
	price: number;
};

export interface PizzaRepository {
	create(input: Omit<Pizza, "id">): Promise<Pizza>;
	getById(id: string): Promise<Pizza | null>;
}

class InMemoryPizzaRepository implements PizzaRepository {
	private readonly store = new Map<string, Pizza>();

	async create(input: Omit<Pizza, "id">): Promise<Pizza> {
		const pizza: Pizza = { id: crypto.randomUUID(), ...input };
		this.store.set(pizza.id, pizza);
		return pizza;
	}

	async getById(id: string): Promise<Pizza | null> {
		return this.store.get(id) ?? null;
	}
}

export const pizzaRepository: PizzaRepository = new InMemoryPizzaRepository();
